File: code/java/ch15/practice/Warmup01PairSumSorted.java
package ch15.practice;

import java.util.*;

/**
 * Warmup 1: Pair Sum in Sorted Array
 * Chapter 15: Two Pointers & Sliding Window — The Caterpillar Method
 *
 * PROBLEM: Given a sorted array and target, find two numbers summing to target.
 *          Return [a, b] with a <= b, smallest first element. Return [-1,-1] if none.
 *
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */
public class Warmup01PairSumSorted {
    public static int[] solve(int[] arr, int target) {
        // TODO: Replace this with your solution
        return new int[]{-1, -1};
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String line = sc.nextLine().trim();
        int[] arr = line.isEmpty() ? new int[]{} :
            Arrays.stream(line.split(" ")).mapToInt(Integer::parseInt).toArray();
        int target = sc.nextInt();
        System.out.println(Arrays.toString(solve(arr, target)));
        sc.close();
    }
}

File: code/cpp/ch15/practice/warmup_01_pair_sum_sorted.cpp
/*
 * Warmup 1: Pair Sum in Sorted Array
 * Chapter 15: Two Pointers & Sliding Window — The Caterpillar Method
 *
 * PROBLEM: Given a sorted array and target, find pair summing to target.
 *          Return {a, b} with a <= b. Return {-1, -1} if none.
 */

#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;

vector<int> solve(vector<int> arr, int target) {
    // TODO: Replace this with your solution
    return {-1, -1};
}

int main() {
    int n;
    cin >> n;
    vector<int> arr(n);
    for (int i = 0; i < n; i++) cin >> arr[i];
    int target;
    cin >> target;
    vector<int> result = solve(arr, target);
    cout << result[0] << " " << result[1] << endl;
    return 0;
}
